src/Language/PureScript/Errors.hs

```diff
@@ -1195,6 +1196,14 @@ renderBox = unlines
   dropWhileEnd p = reverse . dropWhile p . reverse
   whiteSpace = all isSpace
 
+toTypelevelString :: Type -> Maybe Box.Box
+toTypelevelString (TypeLevelString s) = Just $ Box.text s
+toTypelevelString (TypeApp (TypeConstructor f) x)
+  | f == primName "TypeString" = Just $ typeAsBox x
+toTypelevelString (TypeApp (TypeApp (TypeConstructor f) x) ret)
+  | f == primName "TypeConcat" = (Box.<>) <$> (toTypelevelString x) <*> (toTypelevelString ret)
```

Comment:
Instead of `<>`, we can use `before`, which will correctly handle boxes with multiple lines, like some types.